File: src/brs/http/SuggestFee.java
package brs.http;

import brs.feesuggestions.FeeSuggestion;
import brs.feesuggestions.FeeSuggestionCalculator;
import org.json.simple.JSONObject;
import org.json.simple.JSONStreamAware;

import javax.servlet.http.HttpServletRequest;

import static brs.http.common.ResultFields.*;

public class SuggestFee extends APIServlet.APIRequestHandler {

  private final FeeSuggestionCalculator feeSuggestionCalculator;

  public SuggestFee(FeeSuggestionCalculator feeSuggestionCalculator) {
    super(new APITag[]{APITag.FEES});
    this.feeSuggestionCalculator = feeSuggestionCalculator;
  }

  @Override
  JSONStreamAware processRequest(HttpServletRequest req) {
    final FeeSuggestion feeSuggestion = feeSuggestionCalculator.giveFeeSuggestion();

    final JSONObject response = new JSONObject();

    response.put(CHEAP_FEE_RESPONSE, feeSuggestion.getCheapFee());
    response.put(STANDARD_FEE_RESPONSE, feeSuggestion.getStandardFee());
    response.put(PRIORITY_FEE_RESPONSE, feeSuggestion.getPriorityFee());

    return response;
  }

}

File: src/brs/http/SuggestFee.kt
package brs.http

import brs.feesuggestions.FeeSuggestion
import brs.feesuggestions.FeeSuggestionCalculator
import com.google.gson.JsonElement
import com.google.gson.JsonObject

import javax.servlet.http.HttpServletRequest

import brs.http.common.ResultFields.*

internal class SuggestFee(private val feeSuggestionCalculator: FeeSuggestionCalculator) : APIServlet.JsonRequestHandler(arrayOf(APITag.FEES)) {

    internal override fun processRequest(req: HttpServletRequest): JsonElement {
        val feeSuggestion = feeSuggestionCalculator.giveFeeSuggestion()

        val response = JsonObject()

        response.addProperty(CHEAP_FEE_RESPONSE, feeSuggestion.cheapFee)
        response.addProperty(STANDARD_FEE_RESPONSE, feeSuggestion.standardFee)
        response.addProperty(PRIORITY_FEE_RESPONSE, feeSuggestion.priorityFee)

        return response
    }

}
